File: lesson-03/java/ex07.java
public class ex07 {
    public static void main(String[] args) {
        int res = fact(5);
        System.out.println(res);
    }

    public static int fact(int n) {
        return (n == 0) ? 1 : n * fact(n - 1);
    }
}

File: lesson-03/js/ex07.js
// 
// const fact = (n) => {
//     if (n === 0) {
//         return 1;
//     }
//     else {
//         return n * fact(n - 1);
//     }
// }

const fact = (n) => n === 0 ? 1 : n * fact(n - 1)

const res = fact(5);
console.log(res);